pandas/tests/reshape/test_melt.py

```diff
@@ -661,3 +661,36 @@ def test_col_substring_of_stubname(self):
                                  i=['node_id', 'A'],
                                  j='time')
         tm.assert_frame_equal(result, expected)
+
+    def test_melt_missing_columns(self):
+        # GH-23575
+        # This test is to ensure that pandas raises an error if melting is
```

Comment:
This comment could maybe be replaced with a more descriptive test name (though the current name isn't bad). Maybe `test_melt_missing_columns_raises`?